app/models/user.rb

```diff
@@ -18,17 +16,13 @@ class User < ActiveRecord::Base
 
   scope :with_statistics, Queries::StatisticsQuery
 
-  def self.from_omniauth(auth)
-    email_address = auth.info.email
-    grabbed = EMAIL_FORMAT.match(email_address).try(:[], :email)
-    grabbed = grabbed ? "#{grabbed}%" : email_address
-    user = where("email LIKE :email", email: grabbed).first_or_create do |u|
-      u.name= auth.info.name
-      u.email= auth.info.email
-      u.image = auth.info.image
+  def self.from_andela_auth(user)
+    user = where("email LIKE :email", email: user['email']).first_or_create do |u|
+      u.name= user['name']
+      u.email= user['email']
+      u.image = user['picture']
```

Comment:
This was a concern raised earlier, not just the picture, but also the name. One way I can immediately think of is to run a check immediately after if there was a change, run an update else return the user